libvast/builtins/commands/show.cpp

```diff
@@ -40,71 +144,69 @@ caf::message show_command(const invocation& inv, caf::actor_system& sys) {
     return caf::make_message(std::move(components.error()));
   auto [catalog] = std::move(*components);
   // show!
-  auto command_result = caf::message{caf::none};
-  self->request(catalog, caf::infinite, atom::get_v, atom::taxonomies_v)
-    .receive(
-      [&](taxonomies& taxonomies) mutable {
-        auto result = list{};
-        result.reserve(taxonomies.concepts.size());
-        if (inv.full_name == "show" || inv.full_name == "show concepts") {
-          for (auto& [name, concept_] : taxonomies.concepts) {
-            auto fields = list{};
-            fields.reserve(concept_.fields.size());
-            for (auto& field : concept_.fields)
-              fields.push_back(std::move(field));
-            auto concepts = list{};
-            concepts.reserve(concept_.concepts.size());
-            for (auto& concept_ : concept_.concepts)
-              concepts.push_back(std::move(concept_));
-            auto entry = record{
-              {"concept",
-               record{
-                 {"name", std::move(name)},
-                 {"description", std::move(concept_.description)},
-                 {"fields", std::move(fields)},
-                 {"concepts", std::move(concepts)},
-               }},
-            };
-            result.push_back(std::move(entry));
+  auto command_result = caf::expected<list>{list{}};
+  if (show_concepts || show_models) {
+    self->request(catalog, caf::infinite, atom::get_v, atom::taxonomies_v)
```

Comment:
Please add an appropriate timeout while you're touching this code.

Reply:
I re-used the existing option we have for this, the node connection timeout.